netbox/netbox/api/serializers/base.py

```diff
@@ -13,7 +13,48 @@
 )
 
 
+class BaseNetBoxHyperlinkedIdentityField(serializers.HyperlinkedIdentityField):
+    """
+    Overrides HyperlinkedIdentityField to use standard NetBox view naming
+    instead of passing in the view_name.  Initialize with a blank view_name
+    and it will get replaced in the get_url call.  Derived classes must
+    define a get_view_name.
+    """
+    def get_url(self, obj, view_name, request, format):
+        """
+        Given an object, return the URL that hyperlinks to the object.
+
+        May raise a `NoReverseMatch` if the `view_name` and `lookup_field`
+        attributes are not configured to correctly match the URL conf.
+        """
+        # Unsaved objects will not yet have a valid URL.
+        if hasattr(obj, 'pk') and obj.pk in (None, ''):
+            return None
+
+        lookup_value = getattr(obj, self.lookup_field)
+        kwargs = {self.lookup_url_kwarg: lookup_value}
+
+        model_name = self.parent.Meta.model._meta.model_name
+        app_name = self.parent.Meta.model._meta.app_label
+        view_name = self.get_view_name(app_name, model_name)
+        return self.reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class NetBoxAPIHyperlinkedIdentityField(BaseNetBoxHyperlinkedIdentityField):
+
+    def get_view_name(self, app_name, model_name):
+        return f'{app_name}-api:{model_name}-detail'
+
+
+class NetBoxURLHyperlinkedIdentityField(BaseNetBoxHyperlinkedIdentityField):
+
+    def get_view_name(self, app_name, model_name):
+        return f'{app_name}:{model_name}'
+
+
 class BaseModelSerializer(serializers.ModelSerializer):
+    url = NetBoxAPIHyperlinkedIdentityField(view_name="")
+    display_url = NetBoxURLHyperlinkedIdentityField(view_name="")
```

Comment:
Since we're using a custom class anyway, let's override its `__init__()` method to avoid having to pass a dummy value for `view_name`.